src/CoreBundle/DependencyInjection/Compiler/ChameleonModulePass.php

```diff
@@ -28,14 +29,16 @@ public function process(ContainerBuilder $container)
     {
         $moduleServiceDefinition = $container->getDefinition('chameleon_system_core.module_resolver');
         $moduleServiceIds = array_keys($container->findTaggedServiceIds('chameleon_system.module'));
+        $services = [];
 
         foreach ($moduleServiceIds as $moduleServiceId) {
             $moduleDefinition = $container->getDefinition($moduleServiceId);
-            if (true === $moduleDefinition->isShared() && ContainerInterface::SCOPE_PROTOTYPE !== $moduleDefinition->getScope()) {
+            if (true === $moduleDefinition->isShared()) {
                 throw new \LogicException('Chameleon modules must not be shared service instances. This module is shared: '.$moduleServiceId);
             }
+            $services[$moduleServiceId] = new Reference($moduleServiceId);
         }
 
-        $moduleServiceDefinition->addMethodCall('addModules', array($moduleServiceIds));
+        $moduleServiceDefinition->replaceArgument(0, ServiceLocatorTagPass::register($container, $services));
```

Comment:
Why is "register()" used as method argument here?

Reply:
See Symfony docs: https://symfony.com/doc/current/service_container/service_subscribers_locators.html#defining-a-service-locator (last example)